src/lib.rs

```diff
@@ -28,21 +28,28 @@ use raw_window_handle::{HasRawDisplayHandle, HasRawWindowHandle, RawDisplayHandl
 /// write to a window on that platform. This struct owns the window that this data corresponds to
 /// to ensure safety, as that data must be destroyed before the window itself is destroyed. You may
 /// access the underlying window via [`window`](Self::window) and [`window_mut`](Self::window_mut).
-pub struct GraphicsContext<W: HasRawWindowHandle + HasRawDisplayHandle> {
-    window: W,
+pub struct GraphicsContext {
     graphics_context_impl: Box<dyn GraphicsContextImpl>,
 }
 
-impl<W: HasRawWindowHandle + HasRawDisplayHandle> GraphicsContext<W> {
-    /// Creates a new instance of this struct, consuming the given window.
+impl GraphicsContext {
+    /// Creates a new instance of this struct, using the provided window.
     ///
     /// # Safety
     ///
-    ///  - Ensure that the passed object is valid to draw a 2D buffer to
-    pub unsafe fn new(window: W) -> Result<Self, SwBufError<W>> {
```

Comment:
Should we necessarily assume the same type implements both `HasRawWindowHandle` and `HasRawDisplayHandle`? That may be true of winit, but perhaps not everything implementing these has both for the same type.

Reply:
Done